bzip2.rs

```diff
@@ -1192,7 +1193,7 @@ unsafe fn set_binary_mode(file: *mut FILE) {
 unsafe fn set_binary_mode(_file: *mut FILE) {}
 
 unsafe fn compress(config: &Config) {
-    delete_output_on_interrupt = false;
+    delete_output_on_interrupt.store(false, Ordering::Relaxed);
```

Comment:
All stores and loads should be `SeqCst` to synchronize with the signal handler. Otherwise I think it is possible for the compiler to delay the store until after the (de)compression is already performed.

Reply:
yeah, looks like it. Though I guess we really should try to get rid of that signal handler (maybe using mio to still catch/respond to the events?). The segfault handler should not be needed once we make the code a bit safer.